discovery-provider/src/queries/get_track_history.py

```diff
@@ -0,0 +1,59 @@
+from src.models import Play, Track
+from src.utils import helpers
+from src.utils.db_session import get_db_read_replica
+from src.queries import response_name_constants
+from src.queries.query_helpers import (
+    add_query_pagination,
+    populate_track_metadata,
+    get_users_by_id,
+    get_users_ids,
+)
+
+
+def get_track_history(args):
+    current_user_id = args.get("current_user_id")
+    limit = args.get("limit")
+    offset = args.get("offset")
+    filter_deleted = args.get("filter_deleted")
+
+    db = get_db_read_replica()
+    with db.scoped_session() as session:
+        base_query = (
+            session.query(Track, Play.created_at)
+            .join(Play, Play.play_item_id == Track.track_id)
+            .filter(
+                Play.user_id == current_user_id,
+                Track.is_current == True,
+            )
+        )
+
+        # Allow filtering of deletes
+        if filter_deleted:
+            base_query = base_query.filter(Track.is_delete == False)
+
+        base_query = base_query.order_by(Play.created_at.desc())
+
+        query_results = add_query_pagination(base_query, limit, offset).all()
+
+        if not query_results:
+            return []
+
+        tracks, save_dates = zip(*query_results)
+        tracks = helpers.query_result_to_list(tracks)
+        track_ids = list(map(lambda track: track["track_id"], tracks))
+
+        # bundle peripheral info into track results
+        tracks = populate_track_metadata(session, track_ids, tracks, current_user_id)
+
+        if args.get("with_users", False):
+            user_id_list = get_users_ids(tracks)
```

Comment:
we don't use this everywhere, but we should:
https://github.com/AudiusProject/audius-protocol/blob/cefbc039c533797ae78584cd5f8c3dc1bbc3841b/discovery-provider/src/queries/get_remixes_of.py#L147

(helper fn)

Reply:
Added